packages/react-search-ui-views/src/view-helpers/getUrlSanitizer.js

```diff
@@ -0,0 +1,29 @@
+const VALID_PROTOCOLS = ["http:", "https:"];
+
+/**
+ *
+ * @param {URL} URLParser URL interface provided by browser https://developer.mozilla.org/en-US/docs/Web/API/URL
+ * @param {String} currentLocation String representation of the browser's current location
+ */
+export default function getUrlSanitizer(URLParser, currentLocation) {
+  // This function is curried so that dependencies can be injected and don't need to be mocked in tests.
+  return url => {
+    let parsedUrl;
+
+    try {
+      // Attempts to parse a URL as absolute
+      parsedUrl = new URLParser(url);
+      // eslint-disable-next-line no-empty
+    } catch (e) {}
+
+    try {
+      // Attempts to parse a URL as relative
+      parsedUrl = new URLParser(url, currentLocation);
+      // eslint-disable-next-line no-empty
+    } catch (e) {}
+
+    if (!parsedUrl) return "";
```

Comment:
Is this if check here necessary? a falsey parsedUrl will still fail the ternary below and return a `""`, right?

Reply:
If this is generally for dev readability however feel free to ignore me and leave this in!

Reply:
You're right, it's no longer necessary. I'd just need to update the ternary below to ensure parsedUrl is not undefined before calling `parsedUrl.protocol`.

```
return VALID_PROTOCOLS.includes((parsedUrl || {}).protocol) ? url : "";
```

Or maybe just change the initial assignment of parsedURL

```
let parsedUrl = {};
```

Reply:
https://github.com/elastic/search-ui/pull/481/commits/c4c9bc75ecddb9b3ac26275bbd8bd9feae3e996a